app/components/Table/index.tsx

```diff
@@ -31,6 +47,44 @@ const Table: React.FC<Props> = ({
     [router]
   );
 
+  const handleRouteUpdate = useCallback(
+    (url, mode: "replace" | "push") => {
+      const afterFetch = () => {
+        setIsRefetching(false);
+        // At this point the data for the query should be cached,
+        // so we can update the route and re-render without suspending
+        if (mode === "replace") router.replace(url, url, { shallow: true });
+        else router.push(url, url, { shallow: true });
+      };
+
+      if (!pageQuery) {
+        afterFetch();
+        return;
+      }
+
+      if (isRefetching) {
+        return;
+      }
+
+      setIsRefetching(true);
+
+      // fetchQuery will fetch the query and write the data to the Relay store.
+      // This will ensure that when we re-render, the data is already cached and we don't suspend
+      // See https://github.com/facebook/relay/blob/b8e78ca0fbbfe05f34b4854484df574d91ba2113/website/docs/guided-tour/refetching/refetching-queries-with-different-data.md#if-you-need-to-avoid-suspense
+      fetchQuery(
+        environment,
+        pageQuery,
+        withRelayOptions.variablesFromContext(url)
+      ).subscribe({
+        complete: afterFetch,
+        // if the query fails, we still want to update the route,
+        // which will retry the query and let a 500 page be rendered if it fails again
+        error: afterFetch,
```

Comment:
I wonder if that will trigger the Sentry boundary before retrying the query? Or is it one of the special cases where we have to call `Sentry.captureException(...)`

Reply:
This isn't an uncaught exception, so Sentry won't be automatically called. I can add `Sentry.captureException` here